osu.Game/Screens/OnlinePlay/Multiplayer/Multiplayer.cs

```diff
@@ -18,8 +19,24 @@ public override void OnResuming(IScreen last)
         {
             base.OnResuming(last);
 
-            if (client.Room != null && client.LocalUser?.State != MultiplayerUserState.Spectating)
-                client.ChangeState(MultiplayerUserState.Idle);
+            if (client.Room == null)
+                return;
+
+            Debug.Assert(client.LocalUser != null);
+
+            switch (client.LocalUser.State)
+            {
+                case MultiplayerUserState.Spectating:
+                    break;
+
+                case MultiplayerUserState.WaitingForLoad:
+                    client.AbortLoad();
+                    break;
+
+                default:
+                    client.ChangeState(MultiplayerUserState.Idle);
```

Comment:
I don't think we want this. As mentioned [here](https://github.com/ppy/osu/issues/14181#issuecomment-990631746), making the method `AbortGameplay` seems like a much better path forward. Having this remaining case feels like it's only a matter of time before it to becomes an issue.

Reply:
There's 5 possible different states:
```
Spectating
WaitingForLoad
Loaded
Playing
Results
```
Which of these would be handled via "AbortGameplay"?

Reply:
The middle three I would say.

ie. any time the user is expected to be in the current gameplay session but chooses to leave.

Reply:
Imo it's a much of a muchness beyond the initial load one, but I've applied the changes. I haven't handled the `FinishedPlay` state which I forgot to mention above.